### PR TITLE
Add support for Private endpoint to SMB file share

### PR DESCRIPTION
**Pull request by shekharsorot**
This PR ports some code from class NFS to class AzureFileShare.
This enables creation of private endpoint, DNS zone and records which is necessary for scenarios where policy does not allows access to SMB share over public endpoint.

note: private endpoint must be created after shares are created. There is no workaround at the moment to do otherwise via API at the moment.

### CODE REVIEW

File: lisa/sut_orchestrator/azure/features.py [hunk 1]
```diff
@@ -3273,6 +3276,55 @@ def create_file_share(
                 resource_group_name=resource_group_name,
                 log=self._log,
             )
+
+        # Create file private endpoint, always after all shares are created
+        if enable_private_endpoint is True:
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/features.py]: `if enable_private_endpoint:`

Comment (shekharsorot) [reply lisa/sut_orchestrator/azure/features.py]: updated code

File: lisa/sut_orchestrator/azure/features.py [hunk 2]
```diff
@@ -3263,6 +3264,8 @@ def create_file_share(
             allow_shared_key_access=allow_shared_key_access,
         )
 
+        # The SMB endpoint https://<share>.file.core.windows.net
+        # will auto point to <share>.privatelink.file.core.windows.net
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/features.py]: The added comments is always true, or only when `enable_private_endpoint` is True? If it happens when `enable_private_endpoint` is true, move the code comment to right place.

Comment (shekharsorot) [reply lisa/sut_orchestrator/azure/features.py]: updated comment to ensure clarity on condition and sequence

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**